bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusMeterHandler.java

```diff
@@ -146,8 +149,11 @@ private void updateData(FroniusBridgeConfiguration bridgeConfiguration, FroniusB
      * @param deviceId of the device
      * @return {MeterRealtimeResponse} the object representation of the json response
      */
-    private MeterRealtimeResponseDTO getMeterRealtimeData(String ip, int deviceId) {
-        String location = FroniusBindingConstants.METER_REALTIME_DATA_URL.replace("%IP%", StringUtils.trimToEmpty(ip));
+    private MeterRealtimeResponseDTO getMeterRealtimeData(final String ip, final Integer deviceId) {
+        Objects.requireNonNull(ip, "IP address must be set in the configuration.");
+        Objects.requireNonNull(deviceId, "Device ID must be set in the configuration.");
```

Comment:
This will throw an unchecked exception, which will crash your binding. You could throw a custom exception and catch it in the calling method.